src/server/models/queries.js

```diff
@@ -0,0 +1,21 @@
+require('dotenv').config();
+const e = require('express');
+const client = require('./db');
+
+const mongourl = process.env.DB_URL;
+const dbName = process.env.DB_DATABASE;
+
+module.exports.getReservesByTerm = async (dbCollection, academicTerm) => {

```

Comment:
@kilometerskang Make sure that only Administrators can call this API

Reply:
How do I do that?

Reply:
@kilometerskang Check the user's roles in idToken: https://cvmcosta.me/ltijs/#/provider?id=idtoken

You did that on the frontend: https://github.com/ucla/lti-library-resources/blob/master/src/client/components/App/userRoles.js#L11

Reply:
The AdminPanel doesn't ever get rendered unless the user is an admin. Is that what you mean, or do you mean I should pass in the idToken to the API or something?

Reply:
@kilometerskang A nefarious user just call the API endpoint directly. That's why you also need to secure the backend and not just the frontend. The LTIjs backend should already have the idToken. No need to pass it back.